### ISSUE TITLE
Check if tool_requirements could be necessary

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#451]
dev_requires depend on scopes, but there are build requirements that doesn't affect the hash/packageId, that are required at build time, but not necessarily when dev scope is activated. Suggested by @tru

**Comment by memsharded** [comment]
After checking it, I think that a new type of ``tool_requirements`` could be more complex than just removing them from ``conan_info()``. The worst thing that could happen if someone forget to remove a ``tool_requirement`` is that the binary package ID will change **for major versions**, as the package hash by default uses the semvers of its requirements.

So a tool_requirement like ``CMake`` being upgraded from 3.4 to 3.5 will not generate a different hash in its consumers.

The new interface will be simple:

```python
def conan_info(self):
     self.info.requires.remove("CMake")
```

**Comment by tru** [comment]
Sorry for not following correctly. Are you saying that tool_requirement will be complicated to implement or that emulating it right now is hard and we should add tool_requirement?

**Comment by memsharded** [comment]
I said that it is not necessary to add a new ``tool_requirements`` to the conanfile model. It is almost good as-is:

- Use existing ``requires`` to add tool requirements as cmake.
- This will not change the package binary ID (hash), as long as the major version of cmake it is depending on is changed.
- If the package that consumes ``CMake`` wants to specify that it will not change the binary for any version of cmake, can be remove with the ``conan_info()`` above
- The worst case scenario is to have 2 separated package binaries for different versions of CMake, that are actually the same binary. A bit inefficient but not critical.

Does this clarify what I was trying to tell? Thanks!

**Comment by memsharded** [comment]
This has been addressed by the ``build_requires`` in conan 0.22
